### ISSUE TITLE
Error running documentation example in Jupyter Notebook

### ISSUE THREAD

**Issue by jonhanke** [plastex/plastex#125]
It's my first time using plasTeX so I tried running the basic parsing command below (taken from [https://github.com/plastex/plastex/blob/master/Doc/renderers.tex](https://github.com/plastex/plastex/blob/master/Doc/renderers.tex) lines 131-150, and also on the older sourceforge webpage [here](http://plastex.sourceforge.net/plastex/sect0011.html)), which gives 

    'RuntimeError: generator raised StopIteration`

from the last line `tex.parse()` command (running on Anaconda Python 3).  

Any help resolving this is appreciated!

Thanks,
-Jon

---

<b>Anaconda Python 3 Jupyter Notebook Code Cell:</b>

```
from plasTeX.TeX import TeX

# Instantiate a TeX processor and parse the input text
tex = TeX()
tex.ownerDocument.config['files']['split-level'] = -100
tex.ownerDocument.config['files']['filename'] = 'test.xml'
tex.input(r'''
\documentclass{book}
\begin{document}

Previous paragraph.

\section{My Section}

\begin{center}
Centered text with <, >, and \& charaters.
\end{center}

Next paragraph.

\end{document}
''')
document = tex.parse()
```

<b>Error Message:</b>
```
..
ERROR: An error occurred while building the document object in <tokens> on
   line 1 (generator raised StopIteration)
ERROR: Error while reading argument "name" of documentclass in <string> on
   line 2 (generator raised StopIteration)
ERROR: Error while expanding "documentclass" in <string> on line 2
   (generator raised StopIteration)
ERROR: An error occurred while building the document object in <string> on
   line 2 (generator raised StopIteration)
---------------------------------------------------------------------------
StopIteration                             Traceback (most recent call last)
~/anaconda3/lib/python3.7/site-packages/plasTeX/Tokenizer.py in __iter__(self)
    384             # Get the next character
--> 385             token = next(charIter)
    386 

StopIteration: 

The above exception was the direct cause of the following exception:

RuntimeError                              Traceback (most recent call last)
<ipython-input-138-6e66b06ca27d> in <module>
     15 \end{document}
     16 ''')
---> 17 document = tex.parse()

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in parse(self, output)
    433 
    434         try:
--> 435             for item in tokens:
    436                 if item.nodeType == Macro.ELEMENT_NODE:
    437                     item.parentNode = output

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in __next__(self)
     44         if self._buffer:
     45             return self._buffer.pop()
---> 46         return self._next()
     47     def push(self, value):
     48         self._buffer.append(value)

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in __iter__(self)
    341                     obj.contextDepth = token.contextDepth
    342                     obj.parentNode = token.parentNode
--> 343                     tokens = obj.invoke(self)
    344                     if tokens is None:
    345 #                       log.info('expanding %s %s', token.macroName, obj)

~/anaconda3/lib/python3.7/site-packages/plasTeX/Base/LaTeX/Packages.py in invoke(self, tex)
     34     extension = '.cls'
     35     def invoke(self, tex):
---> 36         a = self.parse(tex)
     37         self.load(tex, a['name'], a['options'])
     38         packages = self.ownerDocument.context.packages

~/anaconda3/lib/python3.7/site-packages/plasTeX/__init__.py in parse(self, tex)
    466                 output, source = tex.readArgumentAndSource(parentNode=self,
    467                                                            name=arg.name,
--> 468                                                            **arg.options)
    469                 self.argSource += source
    470                 self.attributes[arg.name] = output

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in readArgumentAndSource(self, spec, type, subtype, delim, expanded, default, parentNode, name, stripLeadingWhitespace)
    747             # Get a TeX token (i.e. {...})
    748             if spec is None:
--> 749                 toks, source = self.readToken(expanded, parentNode=parentNode)
    750 
    751             # Get a single character argument

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in readToken(self, expanded, parentNode)
    833             # Expand macros and get the argument source string
    834             if expanded:
--> 835                 toks = self.expandTokens(toks, parentNode=parentNode)
    836                 if isgroup:
    837                     s = self.source(toks)

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in expandTokens(self, tokens, normalize, parentNode)
    404         frag = tex.ownerDocument.createDocumentFragment()
    405         frag.parentNode = parentNode
--> 406         out = tex.parse(frag)
    407 
    408         # Pop all of our nested contexts off

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in parse(self, output)
    433 
    434         try:
--> 435             for item in tokens:
    436                 if item.nodeType == Macro.ELEMENT_NODE:
    437                     item.parentNode = output

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in __next__(self)
     44         if self._buffer:
     45             return self._buffer.pop()
---> 46         return self._next()
     47     def push(self, value):
     48         self._buffer.append(value)

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in __iter__(self)
    320         while 1:
    321             # Get the next token
--> 322             token = next()
    323 
    324             # Token is null, ignore it

~/anaconda3/lib/python3.7/site-packages/plasTeX/TeX.py in itertokens(self)
    264             try:
    265                 while 1:
--> 266                     t = next(inputs[-1][-1])
    267                     # Save context depth of each token for use in digestion
    268                     t.contextDepth = context.depth

RuntimeError: generator raised StopIteration
```

**Comment by PatrickMassot** [comment]
Thanks for the report. This is a Python 3.7 incompatibility issue that is fixed in our master branch, but the fix is not yet released on PyPi. I really need to release that fix, more and more people are using Python 3.7. I'd like to incorporate a couple more features and fixes first, but I'll try to work on that really soon.

In the meantime, you can use our master branch by cloning this repository and, from its toplevel folder, run `pip install .`.

**Comment by jonhanke** [comment]
Thanks!  This resolves the issue.
